commit 2f0605a697f4b9f5b1c1571c7ec6a16df4dc2616
Author: Srinivas Dasari <dasaris@codeaurora.org>
Date:   Fri Apr 20 11:41:14 2018 +0530

    nl80211: Free connkeys on external authentication failure
    
    The failure scenario while processing
    NL80211_ATTR_EXTERNAL_AUTH_SUPPORT does not free
    the connkeys. This commit addresses the same.
    
    Signed-off-by: Srinivas Dasari <dasaris@codeaurora.org>
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index ff28f8feeb09..a052693c2e85 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -9214,6 +9214,7 @@ static int nl80211_connect(struct sk_buff *skb, struct genl_info *info)
 
 	if (nla_get_flag(info->attrs[NL80211_ATTR_EXTERNAL_AUTH_SUPPORT])) {
 		if (!info->attrs[NL80211_ATTR_SOCKET_OWNER]) {
+			kzfree(connkeys);
 			GENL_SET_ERR_MSG(info,
 					 "external auth requires connection ownership");
 			return -EINVAL;